### PR TITLE
Okina memory manager segfault fix 

### PR DESCRIPTION
**Pull request by artv3**
Addresses issue #730 .  By default, config.hpp sets 
``int ngpu = -1;`` 
but we have the following method
static inline bool gpuDisabled() { return Get().ngpu == 0; } which will return false if ngpu doesn't get overwritten to 0.  A possible fix is to change the gpuDisabled criteria to 
``static inline bool gpuDisabled() { return Get().ngpu < 0; }``

### CODE REVIEW

File: general/config.hpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ class config
    // **************************************************************************
    static inline void enableGpu(const int dev =0) { Get().MfemDeviceSetup(dev); }
    static inline bool gpuEnabled() { return Get().ngpu > 0; }
-   static inline bool gpuDisabled() { return Get().ngpu == 0; }
+   static inline bool gpuDisabled() { return Get().ngpu < 0; }
    static inline bool gpuHasBeenEnabled() { return Get().ngpu >= 0; }
```
Comment (artv3) [inline general/config.hpp]: @jdahm @camierjs , I don't quite understand the following method. 
``static inline bool gpuHasBeenEnabled() { return Get().ngpu >= 0; }``
Should that check be if ``ngpu >0``  ? Then we can adjust the following method 
``static inline bool gpuDisabled() { return Get().ngpu <= 0; }``

Comment (camierjs) [reply general/config.hpp]: By default, `ngpu` is negative, we also need a state where the gpu has been used, but we are back to CPU mode (laghos uses that for problem 0).

Comment (camierjs) [reply general/config.hpp]: And I'm not sure this is related to the memory manager, is it?

File: general/mm.cpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ void* mm::Insert(void *ptr, const size_t bytes)
 void *mm::Erase(void *ptr)
 {
    if (!config::usingMM()) { return ptr; }
-   if (config::gpuDisabled()) { return ptr; }
+   if (config::gpuDisabled()) { return ptr; } //I think by default the condition should be true.
```
Comment (artv3) [inline general/mm.cpp]: @camierjs , I suspect this is were the problem arises. By default, I think we want ``gpuDisabled()`` to return true. With the existing framework it was returning false. Side note -I added the comment so I can reference that line of code, I wasn't sure how else to do it.

**Comment by jdahm** [comment]
👍

**Pull request merged**